File: migrate_data.py
import sqlite3
import psycopg2

#########
#
# Script to import data from sqlite into postgresql db
#
#########

dbpath = '/srv/data/PiShared/data/bighomedata.db'
#dbpath = 'C:/DATA/PiShared/data/bighomedata.db'
pgcon="host=10.77.0.1 dbname=env_measures  user=user password=pass"

#Postgresql
def GetRecentTimestamp():
    pconn = psycopg2.connect(pgcon)
    pcur = pconn.cursor()
    pcur.execute("SELECT timestamp FROM messwerte order by timestamp desc limit 1;")
    pres=pcur.fetchone()
    pcur.close()
    pconn.close()
    return(pres[0])

def InsertNewMesswerte(messwerte):
    try:
        pconn = psycopg2.connect(pgcon)
        pcur = pconn.cursor()
        pcur.executemany('''INSERT INTO messwerte VALUES(%s,%s,%s,%s,%s)''', messwerte)
        pconn.commit()
        #pcur.close()
        pconn.close()
        res=1
    except Exception as e:
        res=-1
        raise e
    finally:
        return res


def GetNewMesswerte(datum):
    try:
        db = sqlite3.connect(dbpath)
        cursor = db.cursor()
        cursor.execute('''Select distinct timestamp,sensorid,temp ,hum,locationid from messwerte
                  where timestamp >? order by timestamp limit 1000''', (datum,))
        db.commit()
        res=cursor.fetchall()
        db.close()
    except Exception as e:
        res=-1
        raise e
    finally:
        return res

def CountNewMesswerte(datum):
    try:
        db = sqlite3.connect(dbpath)
        cursor = db.cursor()
        cursor.execute('''Select count(distinct timestamp) from messwerte
                  where timestamp >? ''', (datum,))
        db.commit()
        res=cursor.fetchone()
        db.close()
    except Exception as e:
        res=-1
        raise e
    finally:
        return res[0]
#
# Task
tst=GetRecentTimestamp()
#tst='2015-01-10 00:00:00'
#newmes=CountNewMesswerte(tst)
ndat=0
#result=GetNewMesswerte('2015-01-01 00:00:00')
while (CountNewMesswerte(tst)>1):
    result=GetNewMesswerte(tst)
    print(result.__len__())
    insres=InsertNewMesswerte(result)
    ndat=ndat+result.__len__()
    print 'Bis jetzt %d Measures inserted\n' %(ndat)
    tst=GetRecentTimestamp()
    print 'Neuer Zeitpunkt:'+str(tst)
    newmes=CountNewMesswerte(tst)
    print 'Noch %d Messwerte zu verarbeiten' %(newmes)

print(insres)

File: tools/migratedata.R
library (sqldf)

dbfile<-"C:/data/PiShared/data/bighomedata.db"
data <- read.table("c:/data/PiShared/data/klimalog.log", sep="\t",header=TRUE,col.names=c("datum","temp","hum"))


dbdata<-sqldf("select * from messwerte",dbname = dbfile)
locdata<-sqldf("select * from location",dbname = dbfile)

mindat<-sqldf("select min(timestamp) as t from messwerte where sensorid=99",dbname = dbfile)$t

migdata<-sqldf(paste("select datum,99 as sensorid,temp,hum,1 as locationid from data where datum < '",mindat,"'",sep=""))

# transfer<-sqldf("insert into messwerte select datum, sensorid,temp,hum,locationid from migdata",dbname=dbfile)

## Testscenerario
sqldf("create table messwerte_bak as select * from messwerte",dbname = dbfile)

sqldf("select * from messwerte_bak",dbname = dbfile)



transfer<-sqldf("insert into messwerte_bak select datum, sensorid,temp,hum,locationid from migdata",dbname=dbfile)
transfer
sqldf("select * from messwerte_bak",dbname = dbfile)

sqldf("drop table  messwerte_bak",dbname = dbfile)
